client/starwhale/core/runtime/model.py

```diff
@@ -678,6 +705,103 @@ def activate(cls, workdir: str, yaml_name: str) -> None:
+            )
+            conda_export(temp_lock_path, **_kw)
+        else:
+            _py_bin = os.path.join(env_prefix, "bin", "python3")
+            console.print(f":cat_face: use {_py_bin} to freeze requirements...")
+            pip_freeze_by_bin(_py_bin, temp_lock_path, include_editable)
+
+        if stdout:
+            console.rule("requirements lock")
+            with open(temp_lock_path) as f:
+                console.print(f.read())
+            os.unlink(temp_lock_path)
+        else:
+            dest_fname = (
+                RuntimeLockFileType.CONDA
+                if mode == PythonRunEnv.CONDA
+                else RuntimeLockFileType.VENV
+            )
+            console.print(f":mouse: dump lock file: {dest_fname}")
+            shutil.move(temp_lock_path, Path(target_dir) / dest_fname)
+
+            if not disable_auto_inject and runtime_fpath.exists():
+                cls._update_runtime_dep_lock_field(runtime_fpath, dest_fname)
+
+    @staticmethod
+    def _update_runtime_dep_lock_field(runtime_fpath: Path, lock_fname: str) -> None:
+        content = load_yaml(runtime_fpath)
+        deps = content.get("dependencies", [])
+        if lock_fname in deps:
+            return
+
+        console.print(f":monkey_face: update {runtime_fpath} dependencies field")
+        deps.append(lock_fname)
+        content["dependencies"] = deps
+        # TODO: safe_dump will change user's other yaml content format at first time
+        ensure_file(runtime_fpath, yaml.safe_dump(content, default_flow_style=False))
+
+    @staticmethod
+    def _fetch_python_env_meta(runtime_fpath: Path, env: str) -> t.Tuple[str, str, str]:
+        mode, env_prefix, env_name = "", "", ""
+        if runtime_fpath.exists():
+            mode = load_yaml(runtime_fpath).get("mode", PythonRunEnv.VENV)
+
+        if env == RuntimeEnvURIType.SHELL:
+            _current_py_env = guess_current_py_env()
+            if mode and _current_py_env != mode:
+                raise PythonEnvironmentError(
+                    f"runtime.yaml mode:{mode}, shell python env:{_current_py_env}"
+                )
+            mode = mode or _current_py_env
+            env_prefix = get_base_prefix(mode)
+        elif os.path.isdir(env):
+            if os.path.exists(os.path.join(env, "pyvenv.cfg")):
+                _dir_mode = PythonRunEnv.VENV
+            elif os.path.exists(os.path.join(env, "conda-meta")):
+                _dir_mode = PythonRunEnv.CONDA
+            else:
+                _dir_mode = PythonRunEnv.SYSTEM
+
+            if not mode:
+                mode = _dir_mode
+            elif mode != _dir_mode:
+                raise FieldTypeOrValueError(
+                    f"runtime.yaml mode:{mode}, dir mode:{_dir_mode}"
+                )
+
+            env_prefix = env
+        else:
+            if mode == PythonRunEnv.CONDA:
+                env_name = env
```

Comment:
consider `A` as the value of `--env` param
if conda env A exists and path A exists ether, should we report a conflict